### PR TITLE
amp-sidebar: Fix async throw in toolbar

### LINKED ISSUE #27301: Error: Could not find the toolbar-target element with an id: target-element-left
```
Error: Could not find the toolbar-target element with an id: target-element-left
at (https://raw.githubusercontent.com/ampproject/amphtml/2003031842100/extensions/amp-sidebar/0.1/toolbar.js:95)
```

### PR DESCRIPTION
**Pull request by choumx**
Fixes #27301. Saw this as an unlinked error while on release duty.

/to @caroqliu

### CODE REVIEW

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 1]
```diff
@@ -165,14 +165,15 @@ export class AmpSidebar extends AMP.BaseElement {
...
     // Get the toolbar attribute from the child navs.
-    const toolbarElements = toArray(element.querySelectorAll('nav[toolbar]'));
-
-    toolbarElements.forEach(toolbarElement => {
-      try {
-        this.toolbars_.push(new Toolbar(toolbarElement, this));
-      } catch (e) {
-        this.user().error(TAG, 'Failed to instantiate toolbar', e);
-      }
+    this.getAmpDoc().whenReady(() => {
```
Comment (caroqliu) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: `this.getAmpDoc().whenReady().then(() => {})`?

File: extensions/amp-sidebar/0.2/amp-sidebar.js [hunk 2]
```diff
@@ -163,14 +163,15 @@ export class AmpSidebar extends AMP.BaseElement {
...
     // Get the toolbar attribute from the child navs.
-    const toolbarElements = toArray(element.querySelectorAll('nav[toolbar]'));
-
-    toolbarElements.forEach(toolbarElement => {
-      try {
-        this.toolbars_.push(new Toolbar(toolbarElement, this));
-      } catch (e) {
-        this.user().error(TAG, 'Failed to instantiate toolbar', e);
-      }
+    this.getAmpDoc().whenReady(() => {
```
Comment (caroqliu) [inline extensions/amp-sidebar/0.2/amp-sidebar.js]: Same as above

File: extensions/amp-sidebar/0.1/toolbar.js [hunk 3]
```diff
@@ -82,21 +82,18 @@ export class Toolbar {
...
-    });
+    const targetElement = this.ampdoc_.getElementById(targetId);
+    if (targetElement) {
+      this.toolbarTarget_ = targetElement;
+      this.toolbarClone_.classList.add('i-amphtml-toolbar');
+      toggle(this.toolbarTarget_, false);
+    } else {
+      // This error will be later rethrown as a user error and
+      // the side bar will continue to function w/o toolbar feature
+      throw new user().createError(
```
Comment (caroqliu) [inline extensions/amp-sidebar/0.1/toolbar.js]: The comment describes throwing a non-blocking error, which I don't think this was doing before. I am also not sure what "later rethrown" is referring to. Does `createError` ensure the behavior described? I recommend either using `user().warn()` to be able to "continue to function" as described, or placing this as a `userAssert` and removing the legacy comment.

File: extensions/amp-sidebar/0.2/toolbar.js [hunk 4]
```diff
@@ -81,22 +81,18 @@ export class Toolbar {
...
-        );
-      }
-    });
+    const targetElement = this.ampdoc_.getElementById(targetId);
+    if (targetElement) {
+      this.toolbarTarget_ = targetElement;
+      this.toolbarClone_.classList.add('i-amphtml-toolbar');
+      toggle(this.toolbarTarget_, false);
+    } else {
+      // This error will be later rethrown as a user error and
```
Comment (caroqliu) [inline extensions/amp-sidebar/0.2/toolbar.js]: Same as above

File: extensions/amp-sidebar/0.1/toolbar.js (hunk 3 above)
Comment (choumx) [reply extensions/amp-sidebar/0.1/toolbar.js]: I think "later rethrown" refers to the try/catch in amp-sidebar. I'll make this comment less confusing.

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 1 above)
Comment (choumx) [reply extensions/amp-sidebar/0.1/amp-sidebar.js]: Oops, thanks. :)

File: extensions/amp-sidebar/0.1/toolbar.js (hunk 3 above)
Comment (caroqliu) [reply extensions/amp-sidebar/0.1/toolbar.js]: Thanks. Also looks like you want to delete `new` here and in the v2 file.

Comment (choumx) [reply extensions/amp-sidebar/0.1/toolbar.js]: Donezo.

**Pull request merged**